### PR TITLE
Package settings in a struct and reduce code duplication

### PR DESCRIPTION
**Pull request by killenheladagen**
### Checklist before requesting a review

- [X] I have performed a self-review of my own code
- [X] I have verified that the code builds perfectly fine on my local system
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have verified that my code follows the style already available in the repository
- [X] I have made corresponding changes to the documentation

### CODE REVIEW

File: app/dockerdwrapper.c [hunk 1]
```diff
@@ -694,35 +709,15 @@ main(void)
...
-    goto end;
-  }
-  if (!get_ipc_socket_selection(&use_ipc_socket)) {
-    syslog(LOG_INFO, "Failed to get ipc socket selection");
-    exit_code = -1;
-    goto end;
-  }
-
-  if (!start_dockerd(use_sdcard, use_tls, use_ipc_socket)) {
-    syslog(LOG_ERR, "Starting dockerd failed");
```
Comment (madelen-at-work) [inline app/dockerdwrapper.c:725]: this error message got lost?

Comment (killenheladagen) [reply app/dockerdwrapper.c:725]: start_dockerd already reports "Could not execv the dockerd process. Return value: %d, error: %s", but there is also another case with a comment "The process died during adding of callback, tell loop to quit.". I guess I could put this message there.

File: app/dockerdwrapper.c [hunk 2]
```diff
@@ -382,14 +388,33 @@ get_ipc_socket_selection(bool *use_ipc_socket_ret)
...
+  }
+  if (!get_ipc_socket_selection(&settings->use_ipc_socket)) {
+    syslog(LOG_ERR, "Failed to get ipc socket selection");
+    return false;
+  }
+  return true;
+}
+
+// Return true if dockerd was successfully started.
+// Log an error and return false it it failed to start properly.
```
Comment (deepikas20) [inline app/dockerdwrapper.c:410]: Log an error and return false if it failed to start properly - minor typo

Comment (killenheladagen) [inline app/dockerdwrapper.c:410]:
```suggestion
// Log an error and return false if it failed to start properly.
```

**Review by madelen-at-work — APPROVED** [review]

**Review by deepikas20 — APPROVED** [review]

**Pull request merged**